packages/purpose-process/test/purposeService.integration.test.ts

```diff
@@ -65,9 +102,92 @@ describe("database test", async () => {
   });
 
   describe("Purpose service", () => {
-    describe("create purpose", () => {
-      it("TO DO", () => {
-        expect(1).toBe(1);
+    const mockPurpose = getMockPurpose();
+    const mockEService = getMockEService();
+    describe("getPurposeById", () => {
+      it("should get the purpose if it exists", async () => {
+        const mockTenant = {
+          ...getMockTenant(),
+          kind: tenantKind.PA,
+        };
+
+        const mockPurpose1: Purpose = {
+          ...mockPurpose,
+          eserviceId: mockEService.id,
+        };
+        const mockPurpose2: Purpose = {
+          ...getMockPurpose(),
+          id: generateId(),
+          title: "another purpose",
+        };
+        await addOnePurpose(mockPurpose1, postgresDB, purposes);
+        await addOnePurpose(mockPurpose2, postgresDB, purposes);
+        await writeInReadmodel(toReadModelEService(mockEService), eservices);
+        await writeInReadmodel(mockTenant, tenants);
+
+        const result = await purposeService.getPurposeById(
+          mockPurpose1.id,
+          mockTenant.id
+        );
+        expect(result).toMatchObject({
+          purpose: mockPurpose1,
+          isRiskAnalysisValid: false,
+        });
+      });
+      it("should throw purposeNotFound if the purpose doesn't exist", async () => {
+        const notExistingId: PurposeId = generateId();
+        await addOnePurpose(mockPurpose, postgresDB, purposes);
```

Comment:
```suggestion
        await writeInReadmodel(toReadModelEService(mockEService), eservices);
        await writeInReadmodel(mockTenant, tenants);
```
We should set all that's needed from the endpoint minus the thing it should not found, in order to be sure that the problem is, in this case, the missing purpose.

For example if the tenant is retrieved before the purpose this test would fail for missing the tenant

Reply:
I added the tenant in readmodel. The eservice is useless in this case because it would be retrieve based on the purpose.